ablySpec/Auth.swift

```diff
@@ -529,5 +529,256 @@ class Auth : QuickSpec {
                 })
             }
         }
+
+        // RSA10
+        describe("authorise") {
+
+            // RSA10a
+            it("should create a token if needed and use it") {
+                let options = AblyTests.commonAppSetup()
+                var validToken: String?
+
+                // Create a new token
+                waitUntil(timeout: testTimeout) { done in
+                    let rest = ARTRest(options: options)
+
+                    rest.auth.authorise(nil, options: nil, callback: { tokenDetails, error in
+                        expect(rest.auth.method).to(equal(ARTAuthMethod.Token))
+                        expect(tokenDetails).toNot(beNil())
+                        expect(tokenDetails?.token).toNot(beEmpty())
+
+                        if let token = tokenDetails?.token {
+                            validToken = token
+                        }
+
+                        publishTestMessage(rest, completion: { error in
+                            expect(error).to(beNil())
+                            done()
+                        })
+                    })
+                }
+
+                guard let currentToken = validToken else { return }
+
+                waitUntil(timeout: testTimeout) { done in
+                    // New client with Basic auth
```

Comment:
@mattheworiordan I implemented this way because the spec is related just with `Auth#authorise` but I will use the API of the client library instead of `setTokenDetails`.